### PR TITLE
Make the digital::{v1, v2}::InputPin traits proven

### PR DESCRIPTION
**Pull request by nils-van-zuijlen**
See #41 

Previous attempt : #102

### CODE REVIEW

**Comment by nils-van-zuijlen** [comment]
@therealprof I don't understand what you said. Do you mean that the `v1` trait should be proven also?

I did not remove the `unproven` feature flag there because to me it does not make sense to prove a deprecated trait.

And keeping the `v1` trait behind a feature flag forces the `compat` traits to be behind it.

File: src/digital/v2_compat.rs [hunk 1]
```diff
@@ -115,6 +117,7 @@ mod tests {
         }
     }
 
+    #[cfg(feature = "unproven")]
```
Comment (therealprof) [inline src/digital/v2_compat.rs]: You added this.

File: src/digital/v2_compat.rs [hunk 2]
```diff
@@ -140,10 +144,12 @@ mod tests {
...
```
Comment (therealprof) [inline src/digital/v2_compat.rs]: ... and this

File: src/digital/v2_compat.rs [hunk 3]
```diff
@@ -140,10 +144,12 @@ mod tests {
...
     struct NewToggleablePinConsumer<T: v2::ToggleableOutputPin> {
         _pin: T,
     }
 
+    #[cfg(feature = "unproven")]
```
Comment (therealprof) [inline src/digital/v2_compat.rs]: ... and this

File: src/digital/v2_compat.rs [hunk 4]
```diff
@@ -153,6 +159,7 @@ mod tests {
         }
     }
 
+    #[cfg(feature = "unproven")]
```
Comment (therealprof) [inline src/digital/v2_compat.rs]: ... and this

File: src/digital/v2_compat.rs (hunk 1 above)
Comment (nils-van-zuijlen) [reply src/digital/v2_compat.rs]: `v1::StatefulOutputPin` is `unproven`, so this test wouldn't build without the flag anyway.

File: src/digital/v2_compat.rs (hunk 2 above)
Comment (nils-van-zuijlen) [reply src/digital/v2_compat.rs]: `v2::ToggleableOutputPin` is `unproven`, so this test wouldn't build without the flag anyway.

File: src/digital/v2_compat.rs (hunk 3 above)
Comment (nils-van-zuijlen) [reply src/digital/v2_compat.rs]: `v2::ToggleableOutputPin` is `unproven`, and `NewToggleablePinConsumer` depends on it, so this test wouldn't build without the flag anyway.

File: src/digital/v2_compat.rs (hunk 4 above)
Comment (nils-van-zuijlen) [reply src/digital/v2_compat.rs]: same as https://github.com/rust-embedded/embedded-hal/pull/164#discussion_r347004908

**Comment by nils-van-zuijlen** [comment]
I added these `unproven` feature flags to make tests work even when that flag is not enabled.

**Comment by burrbull** [comment]
> I added these `unproven` feature flags to make tests work even when that flag is not enabled.

Remove them.

**Pull request merged**